specs/vep-1243-vdk-meta-jobs/README.md

```diff
@@ -153,6 +188,107 @@ Consider at least the below topics but you do not need to cover those that are n
+3. If there are no errors during step 2., the DAG is built.
+4. The DAG of jobs is executed, taking into account all the order information.
+5. A summary of the DAG is logged.
+
+### Configuration details
+
+You are allowed to specify whether the Meta Job itself should fail if any of its orchestrated jobs encounter errors.
+This is achieved by adding the Data Job attribute **fail_meta_job_on_error**. Set to true, as by default, would mean
+that in the event of a failing orchestrated job, the Meta Job would also fail - with USER error.
+This can help to prevent cascading failures and ensure that issues are isolated and addressed as soon as possible.
+
+There are more settings that are configurable by the user, that can be explored in the
+[config](/projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_configuration.py).
+
+### Capacity Estimation and Constraints
+
+The load to the Control Service APIs from the Meta Jobs plugin mostly depends on three factors:
+
+* Number of Data Jobs in the DAG (there is a reasonable default limit, but the value is
+  [configurable](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_configuration.py#L87);
+* Data Job Executions duration;
+* Configuration value of [META_JOBS_TIME_BETWEEN_STATUS_CHECK_SECONDS](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_configuration.py#L76).
+
+The main resource Meta Jobs consume is API requests. Here are the different cases in which an API request is sent:
+
+* To start a job
+* To check the status of a job (based on META_JOBS_TIME_BETWEEN_STATUS_CHECK_SECONDS)
+* To get details after a job is completed
+* On retries (if a job start failed)
+
+Let's create a formula for calculating the number of API requests in a DAG run. We'll use the following variables:
+
+J: Number of Data Jobs in the DAG
+E: Average Data Job Execution duration
+S: The value of META_JOBS_TIME_BETWEEN_STATUS_CHECK_SECONDS
+
+Generally, job1 is independent of job2 in terms of requests - job1 would not impact the number of requests job2 would
+send throughout its duration.
+
+Thus, 1 job would send (1 + E/S + 1) API request for its run.
+Hence, the DAG would make J * (1 + E/S + 1) total requests.
+
+For example, the following workflow consists of 6 Data Jobs:
+
+* `Job 2`, `Job 3` and `Job 4` depend on `Job 1`;
+* `Job 5` depends on `Job 2` and `Job 3`;
+* `Job 6` depends on `Job 4`;
+* The execution time of each of the following jobs is about 10 min: `Job 1` and `Job 2`;
+* The execution time of each of the following jobs is about 15 min: `Job 3`, `Job 4` and `Job 6`;
+* The execution time of each of the following jobs is about 20 min: `Job 5`;
+* The concurrent running jobs limit is configured to 2.
+* The value of META_JOBS_TIME_BETWEEN_STATUS_CHECK_SECONDS is configured to the default 40 (seconds).
+
+![sample-execution.png](sample-execution.png)
```

Comment:
nit: I find a more obvious example of the DAG capabilities with https://en.wikipedia.org/wiki/Directed_acyclic_graph#/media/File:Tred-G.svg

Reply:
Do you mean that the edges are lying horizontally rather than the standard tree vertical ones? Or you mean that there are some weakly connected components - e.g. a->e, a->c, c->e, while in my example, there aren't such? My idea here was to have a simple DAG and rather focus on the calculations in order to provide some important numbers to the users. I decided to go that way because our existing Meta Jobs example is simple as well and, on the other hand, if the users want to understand better the capabilities of the DAG, they might visit the provided DAG wiki page for such examples.